pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/ImmutableBucket.java

```diff
@@ -155,9 +157,17 @@ void clear(boolean delete) {
         getSnapshotCreateFuture().ifPresent(snapshotGenerateFuture -> {
             if (delete) {
                 snapshotGenerateFuture.cancel(true);
+                String bucketKey = bucketKey();
+                long bucketId = getAndUpdateBucketId();
                 try {
-                    asyncDeleteBucketSnapshot().get(AsyncOperationTimeoutSeconds, TimeUnit.SECONDS);
-                } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                    removeBucketCursorProperty(bucketKey())
+                            .thenApply(__ -> bucketSnapshotStorage.deleteBucketSnapshot(bucketId).exceptionally(ex -> {
+                                log.error("Failed to delete bucket snapshot, bucketId: {}, bucketKey: {}",
+                                        bucketId, bucketKey, ex);
+                                return null;
+                            })).get(AsyncOperationTimeoutSeconds, TimeUnit.SECONDS);
+                } catch (Exception e) {
+                    log.error("Failed to delete bucket snapshot, bucketId: {}, bucketKey: {}", bucketId, bucketKey, e);
```

Comment:
Because `bucketSnapshotStorage.deleteBucketSnapshot` use the same thread as `clear`, so we can't block `deleteBucketSnapshot` when clearing the bucket snapshot.